### DISCUSSION TITLE
Start Using Architecture Decision Records

### DISCUSSION THREAD

**Discussion by iancooper** [BrighterCommand/Brighter#1967] [Ideas]
### Context

It is hard to understand from the codebase why architectural decisions were taken in Brighter - such as use of a single-threaded message pump. This can lead to wasted effort and energy as folks raise PRs to fix "issues" they see, rather than starting by reviewing the existing decision and discussing it

### Decision

Brighter should use [ADRs](https://cognitect.com/blog/2011/11/15/documenting-architecture-decisions) to document its decisions. We should seed this with a number of ADRs to represent existing decisions that it would be helpful to understand.

### Status
Under Discussion

### Consequences
It would be possible to understand the why of Brighter

**Comment by iancooper** [comment]
@preardon @holytshirt @jonnyollifflee Thoughts?

**Reply by DevJonny** [reply]
I like the idea as it helps answer questions like "Why do you it this way?" which we often get to people coming to Brighter for the first time.

**Reply by preardon** [reply]
I am a big fan of ADRs for a variety of reasons, From recalling why we chose something to documentation for the community.

**Comment by holytshirt** [comment]
I love them too!

**Comment by iancooper** [comment]
OK, incoming PR using https://github.com/npryce/adr-tools I'll ping you all as reviewers

**Comment by iancooper** [comment]
LOL, we already have them: https://github.com/BrighterCommand/Brighter/tree/master/doc/adr

**Comment by iancooper** [comment]
OK, that is a bad sign about adoption, but we should use them more

**Comment by iancooper** [comment]
So maybe we should discuss: what ADRs should we backfill with i.e. what do we need to record for understanding where we are?
